bmp.h

```diff
@@ -1,41 +1,75 @@
+// BMP-related data types based on Microsoft's own
+
 #include <cstdint>
 
-typedef uint8_t BYTE;
+/**
+ * Common Data Types
+ *
+ * The data types in this section are essentially aliases for C/C++
+ * primitive data types.
+ *
+ * Adapted from http://msdn.microsoft.com/en-us/library/cc230309.aspx.
+ * See http://en.wikipedia.org/wiki/Stdint.h for more on stdint.h.
+ */
+typedef uint8_t  BYTE;
 typedef uint32_t DWORD;
-typedef int32_t LONG;
+typedef int32_t  LONG;
 typedef uint16_t WORD;
 
+/**
+ * BITMAPFILEHEADER
+ *
+ * The BITMAPFILEHEADER structure contains information about the type, size,
+ * and layout of a file that contains a DIB [device-independent bitmap].
+ *
+ * Adapted from http://msdn.microsoft.com/en-us/library/dd183374(VS.85).aspx.
+ */
+typedef struct
+{
```

Comment:
```suggestion
typedef struct {
```
Follow this bracket-style because the bracket in another line consumes an extra line.